lib/galaxy/model/__init__.py

```diff
@@ -2495,6 +2495,16 @@ def has_data(self):
         """Detects whether there is any data"""
         return self.dataset.has_data()
 
+    def get_created_from_basename(self):
+        return self.dataset.created_from_basename
+
+    def set_created_from_basename(self, created_from_basename):
+        if self.dataset.created_from_basename is not None:
+            raise Exception("Underlying dataset already has a cwlfilename set.")
```

Comment:
Minor, but s/cwlfilename/basename if this is to be used outside the context of the cwl branch?

Reply:
Of course, I rebased this with that fix. Thanks for the catch.